source/source.js

```diff
@@ -42,6 +46,24 @@ var installPromiseMatchers;
 
     $scope.$apply(); // Trigger Promise resolution
 
+    // Trigger $httpBackend flush if any requests are pending
+    try {
+      $httpBackend.flush();
+    } catch (err) {
+      if (err.message !== 'No pending request to flush !') {
```

Comment:
I agree that string matching is not the best approach, but there is no error code on the Error object, and there's no way with $httpBackend or $timeout to find out of there is something to flush without flushing. The good news is, the unit tests do break if the error messages changes. 

Yes, your terms are very reasonable! Thanks for reviewing so quickly.